src/services/signatureHelp.ts

```diff
@@ -29,6 +29,11 @@ namespace ts.SignatureHelp {
             return undefined;
         }
 
+        // In the middle of a string, don't provide signature help unless the user explicitly requested it.
+        if (triggerCharacter !== undefined && isInString(sourceFile, position, startingToken)) {

```

Comment:
what about `,` in an object literal? or `<` in jsxElement?

Reply:
consider extracting this to `isValidTriggerCharacter` function and handling each accordingly.

Reply:
I actually don't think you need to look at the `triggerCharacter` - you just need to use the `startingToken` and check whether you were given a trigger in the first place. That's why in #1150 we had a more abstract idea of a trigger reason rather than the character itself

Reply:
But I think I'd rather handle those cases in another PR.

Reply:
not sure i follow.. why would not it be sufficient, and why anohter PR?